web/services/health_summary_service.go

```diff
@@ -34,38 +34,49 @@ func (s *healthSummaryService) GetHealthSummary() (models.HealthSummary, error)
 	}
 
 	for _, sapSystem := range sapSystems {
-		var clusterIDs []string
 		var hostIDs []string
 
 		for _, instance := range sapSystem.GetAllInstances() {
-			clusterIDs = append(clusterIDs, instance.ClusterID)
 			hostIDs = append(hostIDs, instance.HostID)
 		}
 
-		clusters, err := s.clustersService.GetAll(&ClustersFilter{ID: clusterIDs}, nil)
+		hosts, err := s.hostsService.GetAll(&HostsFilter{ID: hostIDs}, nil)
 		if err != nil {
 			return nil, err
 		}
 
-		hosts, err := s.hostsService.GetAll(&HostsFilter{ID: hostIDs}, nil)
-		if err != nil {
-			return nil, err
+		var hanaClusters []*models.Cluster
+		if sapSystem.AttachedDatabase != nil {
+			var hanaClusterIDs []string
+			for _, instance := range sapSystem.AttachedDatabase.GetAllInstances() {
+				hanaClusterIDs = append(hanaClusterIDs, instance.ClusterID)
```

Comment:
I guess that it won't affect in anything, but here we will be adding the same cluster id 2 times

Reply:
yes it doesnt matter to the filter so i intentionally ignored the duplication